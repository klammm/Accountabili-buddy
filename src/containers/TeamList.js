import React, { Component } from 'react';
import { ScrollView } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { showAllTeams } from '../actions';
import { Button } from '../components/common';
import TeamDetail from '../components/TeamDetail';

// const mapStateToProps = (state) => {
//   return {
//     teamsList: state.teamsList
//   };
// };

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({ showAllTeams }, dispatch);
};

export class TeamList extends Component {
  renderTeams() {
    return this.props.showAllTeams.map(team => <TeamDetail key={team.id} team={team} />)
  }

  render() {
    console.log('all the props in teamslist: ', this.props);
    return (
      <ScrollView>
        {this.renderTeams()}
      </ScrollView>
    );
  }
}

export default connect(mapDispatchToProps)(TeamList);
