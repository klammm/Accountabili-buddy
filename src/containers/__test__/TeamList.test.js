import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { TeamList } from '../TeamList';


describe('tests the TeamList component', () => {
  it('renders TeamList component', () => {
  const component = shallow(
    <TeamList
      showAllTeams={jest.fn}
      renderTeams={jest.fn}
      teamsList={[]}
    />
  );
    expect(toJson(component)).toMatchSnapshot();
  });
});
