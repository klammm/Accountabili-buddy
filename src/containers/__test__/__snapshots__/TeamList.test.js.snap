// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the TeamList component renders TeamList component 1`] = `<ScrollViewMock />`;
